keps/sig-node/3008-qos-class-resources/README.md

```diff
@@ -804,17 +826,28 @@ API.
  type PodSpec struct {
 @@ -3062,6 +3069,10 @@ type PodSpec struct {
         ResourceClaims []PodResourceClaim
-+       // Pod-level resources. Claims, requests and limits are not allowed
-+       // to be specified for pods.
++       // QoSResources specifies the Pod-level requests of QoS resources.
++       // Container-level QoS resources may be specified in which case they
++       // are considered as a default for all containers within the Pod.
++       // +featureGate=QoSResources
 +       // +optional
-+       Resources ResourceRequirements
++       QoSResources []PodQoSResourceRequest
  }
+
++// PodQoSResourceRequest specifies a request for one QoS resource type for a
++// Pod.
++type PodQoSResourceRequest struct {
++       // Name of the QoS resource.
++       Name QoSResourceName
++       // Name of the class (inside the QoS resource type specified by Name field).
++       Class string
++}
 ```
 
-There is already an ongoing effort to add [Pod level resource limits][kep-2837]
-that aims at adding a pod level `Resources` field in a similar fashion. Thus,
-we opt for adding ResourceRequirements insted of QoSResources directly into the
-PodSpec.
+There is an ongoing effort to add [Pod level resource limits][kep-2837] that
+aims at adding a pod level `Resources`. However, we propose to add a distinct
+QoSResources field (with a distinct PodQoSResourceRequest type) in the PodSpec
```

Comment:
I'm not sure I agree with this - can you explain?

Reply:
As per [this comment](https://github.com/kubernetes/enhancements/pull/3004#discussion_r1093864811) I added separate structures for Pod and Container QoS resource requests. Not re-using (or cross-using) any existing types.

Maybe the reference to [KEP-2837](https://github.com/kubernetes/enhancements/pull/1592) is just confusing and should be dropped?